### PR TITLE
Fix #12939: divide-by-0 in Vehicle::UpdateMotionDodgems()

### LINKED ISSUE #12939: [c3ff19] openrct2.exe: Vehicle::UpdateMotionDodgems, Vehicle::UpdateDodgemsMode, Vehicle::UpdateWaitingToDepart, Vehicle::Upd...
<table>
    <tr>
        <td></td>
        <td><b><a href="https://openrct2.sp.backtrace.io/dashboard/openrct2/project/openrct2/group/c3ff19f1b4cfb9e5f7ca906b836ed70f4ea07afae91a8cf041994bd63c7c4cee">New error: [openrct2]</a></b><br /><b>1</b> occurrence since (Fri Nov 15 2019 11:27:39)<br /></td>
    </tr>
</table><table>
    <tr><td><strong>Application</strong></td><td>openrct2.exe</td></tr>
<tr>
    <td><strong>Attributes</strong></td>
    <td>
<strong>dummy</strong> &mdash; 1
        <ul>
                <li><code>0</code> &mdash; 1</li></ul><br>
<strong>attachment_replay.sv6r</strong> &mdash; 1
        <ul>
                <li><code>*</code> &mdash; 1</li></ul><br>
<strong>attachment_park.sv6</strong> &mdash; 1
        <ul>
                <li><code>*</code> &mdash; 1</li></ul><br>
<strong>attachment_config.ini</strong> &mdash; 1
        <ul>
                <li><code>C:\Users\Theel\Documents\OpenRCT2\146bbfda-c3ce-47cc-8682-7b33d7780cf6.ini</code> &mdash; 1</li></ul><br>
<strong>assert_failure</strong> &mdash; 1
        <ul>
                <li><code>*</code> &mdash; 1</li></ul><br>
<strong>attachment_park.sv6.gz</strong> &mdash; 1
        <ul>
                <li><code>C:\Users\Theel\Documents\OpenRCT2\146bbfda-c3ce-47cc-8682-7b33d7780cf6.sv6.gz</code> &mdash; 1</li></ul><br>
<strong>attachment_screenshot.png</strong> &mdash; 1
        <ul>
                <li><code>C:\Users\Theel\Documents\OpenRCT2\screenshot\Do What The Fuck You Want To 2020-09-15 20-11-08.png</code> &mdash; 1</li></ul><br>
<strong>commit</strong> &mdash; 1
        <ul>
                <li><code>135cc10</code> &mdash; 1</li></ul><br></td>
    </tr>
    <tr>
        <td><strong>Classifiers</strong></td>
        <td>
            <strong>intdiv</strong><br></td>
    </tr>
    <tr>
        <td><strong>Frames</strong></td>
        <td>
            <code>
Vehicle::UpdateMotionDodgems
Vehicle::UpdateDodgemsMode
Vehicle::UpdateWaitingToDepart
Vehicle::Update
OpenRCT2::GameState::UpdateLogic</code>
        </td>
    </tr></table>

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Review by tupaschoal — COMMENTED** [review]
Can we take the opportunity to improve the variable naming?

**Comment by janisozaur** [comment]
Why is it 0? Perhaps it just hides the bug

**Comment by IntelOrca** [comment]
Probably because `powered_acceleration` is 0 which is a field from the object file.

**Comment by Gymnasiast** [comment]
ebx is formed by multiplying speed and mass and then bitshifting right by 2. So this can be triggered by having a very low speed or mass.

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -6476,7 +6476,8 @@ int32_t Vehicle::UpdateMotionDodgems()
     }
     _eax -= velocity;
     _eax *= powered_acceleration * 2;
-    _eax /= ebx;
+    if (ebx != 0)
+        _eax /= ebx;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:6480]: we may as well call ebx `momentum` as thats what it represents.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**